From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 15 Oct 2025 20:21:01 -0500
Subject: [PATCH] offload: Fix build failure with --icf=all

Origin: vendor
Forwarded: not-needed
Last-Update: 2026-08-28
---
 openmp/device/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/openmp/device/CMakeLists.txt b/openmp/device/CMakeLists.txt
index a41cc5866938..59509291e4eb 100644
--- a/openmp/device/CMakeLists.txt
+++ b/openmp/device/CMakeLists.txt
@@ -80,7 +80,7 @@ target_compile_options(libompdevice PRIVATE ${compile_options})
 if(NOT "${LLVM_DEFAULT_TARGET_TRIPLE}" MATCHES "^spirv" AND
    NOT "${CMAKE_CXX_COMPILER_TARGET}" MATCHES "^spirv")
   target_link_options(libompdevice PRIVATE
-                  "-flto" "-r" "-nostdlib" "-Wl,--lto-emit-llvm")
+                  "-flto" "-r" "-Wl,--icf=none" "-nostdlib" "-Wl,--lto-emit-llvm")
 else()
   target_link_options(libompdevice PRIVATE
                   "-nostdlib" "-emit-llvm")
